src/rade_api.h

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------*\
+
+  rade_api.h
+
+  Library of API functions that implement the Radio Autoencoder API.
+
+\*---------------------------------------------------------------------------*/
+
+/*
+  Copyright (C) 2024 David Rowe
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions
+  are met:
+
+  - Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  - Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR
+  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef __RADE_API__
+#define __RADE_API__
+
+#include <sys/types.h>
+
+// This declares a single-precision (float) complex number
+
+#ifndef __RADE_COMP__
+#define __RADE_COMP__
+typedef struct {
+  float real;
+  float imag;
+} RADE_COMP;
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Sample rates used
+#define RADE_FS_8000 8000           // modem waveform sample rate
+#define RADE_FS_16000 16000         // speech sample rate
+
+// note single context only in this version, one context has one Tx, and one Rx
+struct rade *rade_open(char model_file[]);
```

Comment:
I know this isn't intended for external users right now but I wonder if we should start using something like Doxygen early on for our own sanity if nothing else.

Reply:
How much work would it be? If its just a few comments in the source I'm OK with it, but wouldn't want to get side tracked.

Is the API really that hard to understand for you?  You are currently the only target user.

Reply:
Yeah, Doxygen is just annotating with comments IIRC. It's been a while since I last used it, though. Regardless, that's something for v2.